src/components/VtkTwoView.vue

```diff
@@ -444,6 +449,27 @@ export default defineComponent({
 
     // --- window/level setup --- //
 
+    const autoRangeValues = computed(() => {
+      // Pre-compute the auto-range values
+      const values: Record<string, Array<number>> = {};
+      if (curImageData?.value) {
+        const scalarData = curImageData.value
+          .getPointData()
+          .getScalars()
+          .getData();
+        const sortedData = [...scalarData].sort(
+          (a: number, b: number) => a - b
+        );
+        Object.entries(WLAutoRanges).forEach(([key, value]) => {
+          const percentage = value * 0.01;
+          const startIdx = Math.round(percentage * sortedData.length);
+          const endIdx = Math.round((1 - percentage) * sortedData.length);
+          values[key] = [sortedData[startIdx], sortedData[endIdx]];
+        });
```

Comment:
Understood! Yes, I was hesitant to leave this approach in place but I did because estimating from the range produced very poor results. For example, the head and neck data (range: [-1024, 3071]):
```
Range       | Actual              | Estimated
--------------------------------------------------
[0.1, 99.9]   [-1019.91, 3066.91]   [-1024, 1446]
[1, 99]       [-983.05, 3030.05]    [-1017, 1045]
[2, 98]       [-942.1, 2989.1]      [-1014, 582]
[5, 95]       [-819.25, 2866.25]    [-1010, 168]
```
The variance between these ranges is also very small and visually not very obvious when toggling between them. Maybe I am missing or misunderstanding something though?

Reply:
Updated to use computed histogram values: [a997bde](https://github.com/Kitware/VolView/pull/442/commits/a997bde3101bf9cd7ea0fa2827d5c9b78effbd85)